### ISSUE TITLE
Bun's question about setting and reading "production" and "development" NODE_ENV variables

### ISSUE THREAD

**Issue by PrinOrange** [oven-sh/bun#9603]
### What version of Bun is running?

1.0.35

### What platform is your computer?

Microsoft Windows NT 10.0.19045.0 x64

### What steps can reproduce the bug?

First,init a bun's project
```bash
bun init
```
And write the code in index.ts
```ts
import process from "process"
console.log(process.env.NODE_ENV)
```
Then follow the [tutorial on deploying to the production environment in the documentation](https://bun.sh/docs/bundler/executables#deploying-to-production), execute such command
```bash
bun build --compile --minify --sourcemap index.ts --outfile myapp
./myapp
```

And it prints
```
undefined
```

### What is the expected behavior?

If it is deployed to a production environment, should it outputs `production` ?

### What do you see instead?

It outputs `undefined` rather than `production`

### Additional information

Besides that, I also want to know, how to execute a script with `bun run` in production mode?
If I execute the above code with `bun run index.ts`, and it prints `development`.

This means that every time I use `bun run` to directly execute a script file, it will be executed in the NODE_ENV = development environment.
But what should I do so that the script can be executed in the environment of NODE_ENV = production? 
Or, is bun run a script specially set up for the development environment? How can I directly execute the script in production mode?

**Comment by PrinOrange** [comment]
I think, is it possible to add a startup parameter after `bun run` to select the running environment? 
For example 
```
bun run index.ts --production
bun run index.ts --development
```

**Comment by dunadain** [comment]
NODE_ENV must be set manually.
try this
NODE_ENV=production bun index.ts

**Comment by PrinOrange** [comment]
This Problem was solved in the version 1.0.36

**Comment by Jarred-Sumner** [comment]
@PrinOrange this was a regression in Bun v1.0.35 - sorry about that
